nucliadb/nucliadb/search/api/v1/chat.py

```diff
@@ -142,42 +138,48 @@ async def generate_answer(
         kbid: str,
         predict: PredictEngine,
         chat_model: ChatModel,
+        features: List[ChatOptions],
     ):
-        bytes_results = base64.b64encode(results.json().encode())
-        yield len(bytes_results).to_bytes(length=4, byteorder="big", signed=False)
-        yield bytes_results
+        if ChatOptions.PARAGRAPHS in features:
+            bytes_results = base64.b64encode(results.json().encode())
+            yield len(bytes_results).to_bytes(length=4, byteorder="big", signed=False)
+            yield bytes_results
 
         answer = []
         async for data in predict.chat_query(kbid, chat_model):
             answer.append(data)
             yield data
-        yield END_OF_STREAM
-
-        text_answer = b"".join(answer)
-
-        detected_entities = await predict.detect_entities(kbid, text_answer.decode())
-        relation_request = RelationSearchRequest()
-        relation_request.subgraph.entry_points.extend(detected_entities)
-        relation_request.subgraph.depth = 1
-
-        relations_results: List[RelationSearchResponse]
-        (
-            relations_results,
-            incomplete_results,
-            queried_nodes,
-            queried_shards,
-        ) = await node_query(kbid, Method.RELATIONS, relation_request, item.shards)
-        yield base64.b64encode(
+
+        if ChatOptions.RELATIONS in features:
+            yield END_OF_STREAM
```

Comment:
should this be nested under the `if ChatOptions.RELATIONS in features:` condition?

Reply:
If you do not ask for relations the end of stream is the end of the stream :)